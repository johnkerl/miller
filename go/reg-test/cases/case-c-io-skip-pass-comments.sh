# TODO: pending --ofmt

run_mlr --skip-comments --idkvp --oxtab cat $indir/comments/comments1.dkvp
run_mlr --pass-comments --idkvp --oxtab cat $indir/comments/comments1.dkvp
run_mlr --skip-comments --idkvp --oxtab cat $indir/comments/comments2.dkvp
run_mlr --pass-comments --idkvp --oxtab cat $indir/comments/comments2.dkvp
run_mlr --skip-comments --idkvp --oxtab cat $indir/comments/comments3.dkvp
run_mlr --pass-comments --idkvp --oxtab cat $indir/comments/comments3.dkvp
# It's annoying trying to check in text files (especially CSV) with CRLF
# to Git, given that it likes to 'fix' line endings for multi-platform use.
# It's easy to simply create CRLF on the fly.
run_mlr_no_output termcvt --lf2crlf < $indir/comments/comments1.dkvp > $outdir/comments1-crlf.dkvp
run_mlr --skip-comments --idkvp --oxtab cat $outdir/comments1-crlf.dkvp
run_mlr --pass-comments --idkvp --oxtab cat $outdir/comments1-crlf.dkvp

run_mlr --skip-comments-with @@ --idkvp --oxtab cat $indir/comments/comments1-atat.dkvp
run_mlr --pass-comments-with @@ --idkvp --oxtab cat $indir/comments/comments1-atat.dkvp
run_mlr --skip-comments-with @@ --idkvp --oxtab cat $indir/comments/comments2-atat.dkvp
run_mlr --pass-comments-with @@ --idkvp --oxtab cat $indir/comments/comments2-atat.dkvp
run_mlr --skip-comments-with @@ --idkvp --oxtab cat $indir/comments/comments3-atat.dkvp
run_mlr --pass-comments-with @@ --idkvp --oxtab cat $indir/comments/comments3-atat.dkvp

run_mlr --skip-comments --inidx --oxtab cat $indir/comments/comments1.nidx
run_mlr --pass-comments --inidx --oxtab cat $indir/comments/comments1.nidx
run_mlr --skip-comments --inidx --oxtab cat $indir/comments/comments2.nidx
run_mlr --pass-comments --inidx --oxtab cat $indir/comments/comments2.nidx
run_mlr --skip-comments --inidx --oxtab cat $indir/comments/comments3.nidx
run_mlr --pass-comments --inidx --oxtab cat $indir/comments/comments3.nidx
# It's annoying trying to check in text files (especially CSV) with CRLF
# to Git, given that it likes to 'fix' line endings for multi-platform use.
# It's easy to simply create CRLF on the fly.
run_mlr_no_output termcvt --lf2crlf < $indir/comments/comments1.nidx > $outdir/comments1-crlf.nidx
run_mlr --skip-comments --inidx --oxtab cat $outdir/comments1-crlf.nidx
run_mlr --pass-comments --inidx --oxtab cat $outdir/comments1-crlf.nidx

run_mlr --skip-comments --ijson --odkvp cat $indir/comments/comments1.json
run_mlr --pass-comments --ijson --odkvp cat $indir/comments/comments1.json
run_mlr --skip-comments --ijson --odkvp cat $indir/comments/comments2.json
run_mlr --pass-comments --ijson --odkvp cat $indir/comments/comments2.json
run_mlr --skip-comments --ijson --odkvp cat $indir/comments/comments3.json
run_mlr --pass-comments --ijson --odkvp cat $indir/comments/comments3.json
# It's annoying trying to check in text files (especially CSV) with CRLF
# to Git, given that it likes to 'fix' line endings for multi-platform use.
# It's easy to simply create CRLF on the fly.
run_mlr_no_output termcvt --lf2crlf < $indir/comments/comments1.json > $outdir/comments1-crlf.json
run_mlr --skip-comments --ijson --odkvp cat $outdir/comments1-crlf.json
run_mlr --pass-comments --ijson --odkvp cat $outdir/comments1-crlf.json

run_mlr --skip-comments --ixtab --odkvp cat $indir/comments/comments1.xtab
run_mlr --pass-comments --ixtab --odkvp cat $indir/comments/comments1.xtab
run_mlr --skip-comments --ixtab --odkvp cat $indir/comments/comments2.xtab
run_mlr --pass-comments --ixtab --odkvp cat $indir/comments/comments2.xtab
# It's annoying trying to check in text files (especially CSV) with CRLF
# to Git, given that it likes to 'fix' line endings for multi-platform use.
# It's easy to simply create CRLF on the fly.
run_mlr_no_output termcvt --lf2crlf < $indir/comments/comments1.xtab > $outdir/comments1-crlf.xtab
run_mlr --skip-comments --ixtab --odkvp cat $outdir/comments1-crlf.xtab
run_mlr --pass-comments --ixtab --odkvp cat $outdir/comments1-crlf.xtab

run_mlr --skip-comments --icsvlite --odkvp cat $indir/comments/comments1.csv
run_mlr --pass-comments --icsvlite --odkvp cat $indir/comments/comments1.csv
run_mlr --skip-comments --icsvlite --odkvp cat $indir/comments/comments2.csv
run_mlr --pass-comments --icsvlite --odkvp cat $indir/comments/comments2.csv
# It's annoying trying to check in text files (especially CSV) with CRLF
# to Git, given that it likes to 'fix' line endings for multi-platform use.
# It's easy to simply create CRLF on the fly.
run_mlr_no_output termcvt --lf2crlf < $indir/comments/comments1.csv > $outdir/comments1-crlf.csv
run_mlr --skip-comments --icsvlite --odkvp cat $outdir/comments1-crlf.csv
run_mlr --pass-comments --icsvlite --odkvp cat $outdir/comments1-crlf.csv

run_mlr --skip-comments --icsv --odkvp cat $indir/comments/comments1.csv
run_mlr --pass-comments --icsv --odkvp cat $indir/comments/comments1.csv
run_mlr --skip-comments --icsv --odkvp cat $indir/comments/comments2.csv
run_mlr --pass-comments --icsv --odkvp cat $indir/comments/comments2.csv
# It's annoying trying to check in text files (especially CSV) with CRLF
# to Git, given that it likes to 'fix' line endings for multi-platform use.
# It's easy to simply create CRLF on the fly.
run_mlr_no_output termcvt --lf2crlf < $indir/comments/comments1.csv > $outdir/comments1-crlf.csv
run_mlr --skip-comments --icsv --odkvp cat $outdir/comments1-crlf.csv
run_mlr --pass-comments --icsv --odkvp cat $outdir/comments1-crlf.csv

run_mlr --oxtab put '$c=$a;$d=$b;$e=hexfmt($a);$f=hexfmt($b)' $indir/int64io.dkvp
# There is different rounding on i386 vs. x86_64 but the essential check is
# that overflow has been avoided. Hence the %g output format here.
run_mlr --oxtab --ofmt '%.8g' put '$p0=$p+0;$p1=$p+1;$p2=$p+2;$p3=$p+3' $indir/int64arith.dkvp
run_mlr --oxtab --ofmt '%.8g' put '$p0=$p-0;$p1=$p-1;$p2=$p-2;$p3=$p-3' $indir/int64arith.dkvp
run_mlr --oxtab --ofmt '%.8g' put '$p0=$p*0;$p1=$p*1;$p2=$p*2;$p3=$p*3' $indir/int64arith.dkvp
run_mlr --oxtab --ofmt '%.8g' put '$n0=$n+0;$n1=$n+1;$n2=$n+2;$n3=$n+3' $indir/int64arith.dkvp
run_mlr --oxtab --ofmt '%.8g' put '$n0=$n-0;$n1=$n-1;$n2=$n-2;$n3=$n-3' $indir/int64arith.dkvp
run_mlr --oxtab --ofmt '%.8g' put '$n0=$n*0;$n1=$n*1;$n2=$n*2;$n3=$n*3' $indir/int64arith.dkvp
