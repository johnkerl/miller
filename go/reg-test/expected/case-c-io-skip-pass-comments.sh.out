
mlr --skip-comments --idkvp --oxtab cat ./reg-test/input/comments/comments1.dkvp
a 1
b 2
c 3

a 4
b 5
c 6

mlr --pass-comments --idkvp --oxtab cat ./reg-test/input/comments/comments1.dkvp
# hello world 1
a 1
b 2
c 3

a 4
b 5
c 6

mlr --skip-comments --idkvp --oxtab cat ./reg-test/input/comments/comments2.dkvp
a 1
b 2
c 3

a 4
b 5
c 6

mlr --pass-comments --idkvp --oxtab cat ./reg-test/input/comments/comments2.dkvp
a 1
b 2
c 3
# hello world 2

a 4
b 5
c 6

mlr --skip-comments --idkvp --oxtab cat ./reg-test/input/comments/comments3.dkvp
a 1
b 2
c 3

a 4
b 5
c 6

mlr --pass-comments --idkvp --oxtab cat ./reg-test/input/comments/comments3.dkvp
a 1
b 2
c 3

a 4
b 5
c 6
# hello world 3


mlr --skip-comments --idkvp --oxtab cat /Users/kerl/pub_http_internet/miller-releases/miller-head/go/output-reg-test/comments1-crlf.dkvp
a 1
b 2
c 3

a 4
b 5
c 6

mlr --pass-comments --idkvp --oxtab cat /Users/kerl/pub_http_internet/miller-releases/miller-head/go/output-reg-test/comments1-crlf.dkvp
# hello world 1
a 1
b 2
c 3

a 4
b 5
c 6

mlr --skip-comments-with @@ --idkvp --oxtab cat ./reg-test/input/comments/comments1-atat.dkvp
a 1
b 2
c 3

a 4
b 5
c 6

mlr --pass-comments-with @@ --idkvp --oxtab cat ./reg-test/input/comments/comments1-atat.dkvp
@@ hello world 1
a 1
b 2
c 3

a 4
b 5
c 6

mlr --skip-comments-with @@ --idkvp --oxtab cat ./reg-test/input/comments/comments2-atat.dkvp
a 1
b 2
c 3

a 4
b 5
c 6

mlr --pass-comments-with @@ --idkvp --oxtab cat ./reg-test/input/comments/comments2-atat.dkvp
a 1
b 2
c 3
@@ hello world 2

a 4
b 5
c 6

mlr --skip-comments-with @@ --idkvp --oxtab cat ./reg-test/input/comments/comments3-atat.dkvp
a 1
b 2
c 3

a 4
b 5
c 6

mlr --pass-comments-with @@ --idkvp --oxtab cat ./reg-test/input/comments/comments3-atat.dkvp
a 1
b 2
c 3

a 4
b 5
c 6
@@ hello world 3

mlr --skip-comments --inidx --oxtab cat ./reg-test/input/comments/comments1.nidx
1 aX1
2 bX2
3 cX3

1 aX4
2 bX5
3 cX6

mlr --pass-comments --inidx --oxtab cat ./reg-test/input/comments/comments1.nidx
# hello world 1
1 aX1
2 bX2
3 cX3

1 aX4
2 bX5
3 cX6

mlr --skip-comments --inidx --oxtab cat ./reg-test/input/comments/comments2.nidx
1 aX1
2 bX2
3 cX3

1 aX4
2 bX5
3 cX6

mlr --pass-comments --inidx --oxtab cat ./reg-test/input/comments/comments2.nidx
1 aX1
2 bX2
3 cX3
# hello world 2

1 aX4
2 bX5
3 cX6

mlr --skip-comments --inidx --oxtab cat ./reg-test/input/comments/comments3.nidx
1 aX1
2 bX2
3 cX3

1 aX4
2 bX5
3 cX6

mlr --pass-comments --inidx --oxtab cat ./reg-test/input/comments/comments3.nidx
1 aX1
2 bX2
3 cX3

1 aX4
2 bX5
3 cX6
# hello world 3


mlr --skip-comments --inidx --oxtab cat /Users/kerl/pub_http_internet/miller-releases/miller-head/go/output-reg-test/comments1-crlf.nidx
1 aX1
2 bX2
3 cX3

1 aX4
2 bX5
3 cX6

mlr --pass-comments --inidx --oxtab cat /Users/kerl/pub_http_internet/miller-releases/miller-head/go/output-reg-test/comments1-crlf.nidx
# hello world 1
1 aX1
2 bX2
3 cX3

1 aX4
2 bX5
3 cX6

mlr --skip-comments --ijson --odkvp cat ./reg-test/input/comments/comments1.json
a=1,b=2,c=3
a=4,b=5,c=6

mlr --pass-comments --ijson --odkvp cat ./reg-test/input/comments/comments1.json
# hello world 1
a=1,b=2,c=3
a=4,b=5,c=6

mlr --skip-comments --ijson --odkvp cat ./reg-test/input/comments/comments2.json
a=1,b=2,c=3
a=4,b=5,c=6

mlr --pass-comments --ijson --odkvp cat ./reg-test/input/comments/comments2.json
# hello world 2
a=1,b=2,c=3
a=4,b=5,c=6

mlr --skip-comments --ijson --odkvp cat ./reg-test/input/comments/comments3.json
a=1,b=2,c=3
a=4,b=5,c=6

mlr --pass-comments --ijson --odkvp cat ./reg-test/input/comments/comments3.json
# hello world 3
a=1,b=2,c=3
a=4,b=5,c=6


mlr --skip-comments --ijson --odkvp cat /Users/kerl/pub_http_internet/miller-releases/miller-head/go/output-reg-test/comments1-crlf.json
a=1,b=2,c=3
a=4,b=5,c=6

mlr --pass-comments --ijson --odkvp cat /Users/kerl/pub_http_internet/miller-releases/miller-head/go/output-reg-test/comments1-crlf.json
# hello world 1
a=1,b=2,c=3
a=4,b=5,c=6

mlr --skip-comments --ixtab --odkvp cat ./reg-test/input/comments/comments1.xtab
x=1,z=3
x=2,z=5

mlr --pass-comments --ixtab --odkvp cat ./reg-test/input/comments/comments1.xtab
#hello world3
x=1,z=3
#hello world4
x=2,z=5

mlr --skip-comments --ixtab --odkvp cat ./reg-test/input/comments/comments2.xtab
x=1,z=3
x=2,z=5

mlr --pass-comments --ixtab --odkvp cat ./reg-test/input/comments/comments2.xtab
#hello world1
#hello world2
#hello world3
x=1,z=3
#hello world4
#hello world5
x=2,z=5
#hello world6


mlr --skip-comments --ixtab --odkvp cat /Users/kerl/pub_http_internet/miller-releases/miller-head/go/output-reg-test/comments1-crlf.xtab
x=1,z=3
x=2,z=5

mlr --pass-comments --ixtab --odkvp cat /Users/kerl/pub_http_internet/miller-releases/miller-head/go/output-reg-test/comments1-crlf.xtab
#hello world3
x=1,z=3
#hello world4
x=2,z=5

mlr --skip-comments --icsvlite --odkvp cat ./reg-test/input/comments/comments1.csv
a=1,b=2,c=3
a=4,b=5,c=6

mlr --pass-comments --icsvlite --odkvp cat ./reg-test/input/comments/comments1.csv
# hello
a=1,b=2,c=3
a=4,b=5,c=6

mlr --skip-comments --icsvlite --odkvp cat ./reg-test/input/comments/comments2.csv
a=1,b=2,c=3
a=4,b=5,c=6

mlr --pass-comments --icsvlite --odkvp cat ./reg-test/input/comments/comments2.csv
# hello
# hi
a=1,b=2,c=3
# there
a=4,b=5,c=6
# haha


mlr --skip-comments --icsvlite --odkvp cat /Users/kerl/pub_http_internet/miller-releases/miller-head/go/output-reg-test/comments1-crlf.csv
a=1,b=2,c=3
a=4,b=5,c=6

mlr --pass-comments --icsvlite --odkvp cat /Users/kerl/pub_http_internet/miller-releases/miller-head/go/output-reg-test/comments1-crlf.csv
# hello
a=1,b=2,c=3
a=4,b=5,c=6

mlr --skip-comments --icsv --odkvp cat ./reg-test/input/comments/comments1.csv
a=1,b=2,c=3
a=4,b=5,c=6

mlr --pass-comments --icsv --odkvp cat ./reg-test/input/comments/comments1.csv
# hello
a=1,b=2,c=3
a=4,b=5,c=6

mlr --skip-comments --icsv --odkvp cat ./reg-test/input/comments/comments2.csv
a=1,b=2,c=3
a=4,b=5,c=6

mlr --pass-comments --icsv --odkvp cat ./reg-test/input/comments/comments2.csv
# hello
# hi
a=1,b=2,c=3
# there
a=4,b=5,c=6
# haha


mlr --skip-comments --icsv --odkvp cat /Users/kerl/pub_http_internet/miller-releases/miller-head/go/output-reg-test/comments1-crlf.csv
a=1,b=2,c=3
a=4,b=5,c=6

mlr --pass-comments --icsv --odkvp cat /Users/kerl/pub_http_internet/miller-releases/miller-head/go/output-reg-test/comments1-crlf.csv
# hello
a=1,b=2,c=3
a=4,b=5,c=6

mlr --oxtab put $c=$a;$d=$b;$e=hexfmt($a);$f=hexfmt($b) ./reg-test/input/int64io.dkvp
a 0x1
b 1
c 1
d 1
e 0x1
f 0x1

a 0x7ffffffffffffffe
b 9223372036854775806
c 9223372036854775806
d 9223372036854775806
e 0x7ffffffffffffffe
f 0x7ffffffffffffffe

a 0x7fffffffffffffff
b 9223372036854775807
c 9223372036854775807
d 9223372036854775807
e 0x7fffffffffffffff
f 0x7fffffffffffffff

a 0x8000000000000000
b -9223372036854775808
c -9223372036854775808
d -9223372036854775808
e 0x8000000000000000
f 0x8000000000000000

a 0xffffffffffffffff
b -1
c -1
d -1
e 0xffffffffffffffff
f 0xffffffffffffffff

mlr --oxtab --ofmt %.8g put $p0=$p+0;$p1=$p+1;$p2=$p+2;$p3=$p+3 ./reg-test/input/int64arith.dkvp
p  0x7ffffffffffff9ff
n  0x8000000000000000
p0 9223372036854774271
p1 9223372036854774272
p2 9223372036854774273
p3 9223372036854774274

p  0x7ffffffffffffa00
n  0x8000000000000001
p0 9223372036854774272
p1 9223372036854774273
p2 9223372036854774274
p3 9223372036854774275

p  0x7ffffffffffffbff
n  0x8000000000000200
p0 9223372036854774783
p1 9223372036854774784
p2 9223372036854774785
p3 9223372036854774786

p  0x7ffffffffffffc00
n  0x8000000000000400
p0 9223372036854774784
p1 9223372036854774785
p2 9223372036854774786
p3 9223372036854774787

p  0x7ffffffffffffe00
n  0x8000000000000401
p0 9223372036854775296
p1 9223372036854775297
p2 9223372036854775298
p3 9223372036854775299

p  0x7ffffffffffffffe
n  0x8000000000000600
p0 9223372036854775806
p1 9223372036854775807
p2 9.223372e+18
p3 9.223372e+18

p  0x7fffffffffffffff
n  0x8000000000000601
p0 9223372036854775807
p1 9.223372e+18
p2 9.223372e+18
p3 9.223372e+18

mlr --oxtab --ofmt %.8g put $p0=$p-0;$p1=$p-1;$p2=$p-2;$p3=$p-3 ./reg-test/input/int64arith.dkvp
p  0x7ffffffffffff9ff
n  0x8000000000000000
p0 9223372036854774271
p1 9223372036854774270
p2 9223372036854774269
p3 9223372036854774268

p  0x7ffffffffffffa00
n  0x8000000000000001
p0 9223372036854774272
p1 9223372036854774271
p2 9223372036854774270
p3 9223372036854774269

p  0x7ffffffffffffbff
n  0x8000000000000200
p0 9223372036854774783
p1 9223372036854774782
p2 9223372036854774781
p3 9223372036854774780

p  0x7ffffffffffffc00
n  0x8000000000000400
p0 9223372036854774784
p1 9223372036854774783
p2 9223372036854774782
p3 9223372036854774781

p  0x7ffffffffffffe00
n  0x8000000000000401
p0 9223372036854775296
p1 9223372036854775295
p2 9223372036854775294
p3 9223372036854775293

p  0x7ffffffffffffffe
n  0x8000000000000600
p0 9223372036854775806
p1 9223372036854775805
p2 9223372036854775804
p3 9223372036854775803

p  0x7fffffffffffffff
n  0x8000000000000601
p0 9223372036854775807
p1 9223372036854775806
p2 9223372036854775805
p3 9223372036854775804

mlr --oxtab --ofmt %.8g put $p0=$p*0;$p1=$p*1;$p2=$p*2;$p3=$p*3 ./reg-test/input/int64arith.dkvp
p  0x7ffffffffffff9ff
n  0x8000000000000000
p0 0
p1 9223372036854774271
p2 1.8446744e+19
p3 2.7670116e+19

p  0x7ffffffffffffa00
n  0x8000000000000001
p0 0
p1 9223372036854774272
p2 1.8446744e+19
p3 2.7670116e+19

p  0x7ffffffffffffbff
n  0x8000000000000200
p0 0
p1 9223372036854774783
p2 1.8446744e+19
p3 2.7670116e+19

p  0x7ffffffffffffc00
n  0x8000000000000400
p0 0
p1 9223372036854774784
p2 1.8446744e+19
p3 2.7670116e+19

p  0x7ffffffffffffe00
n  0x8000000000000401
p0 0
p1 9.223372e+18
p2 1.8446744e+19
p3 2.7670116e+19

p  0x7ffffffffffffffe
n  0x8000000000000600
p0 0
p1 9.223372e+18
p2 1.8446744e+19
p3 2.7670116e+19

p  0x7fffffffffffffff
n  0x8000000000000601
p0 0
p1 9.223372e+18
p2 1.8446744e+19
p3 2.7670116e+19

mlr --oxtab --ofmt %.8g put $n0=$n+0;$n1=$n+1;$n2=$n+2;$n3=$n+3 ./reg-test/input/int64arith.dkvp
p  0x7ffffffffffff9ff
n  0x8000000000000000
n0 -9223372036854775808
n1 -9223372036854775807
n2 -9223372036854775806
n3 -9223372036854775805

p  0x7ffffffffffffa00
n  0x8000000000000001
n0 -9223372036854775807
n1 -9223372036854775806
n2 -9223372036854775805
n3 -9223372036854775804

p  0x7ffffffffffffbff
n  0x8000000000000200
n0 -9223372036854775296
n1 -9223372036854775295
n2 -9223372036854775294
n3 -9223372036854775293

p  0x7ffffffffffffc00
n  0x8000000000000400
n0 -9223372036854774784
n1 -9223372036854774783
n2 -9223372036854774782
n3 -9223372036854774781

p  0x7ffffffffffffe00
n  0x8000000000000401
n0 -9223372036854774783
n1 -9223372036854774782
n2 -9223372036854774781
n3 -9223372036854774780

p  0x7ffffffffffffffe
n  0x8000000000000600
n0 -9223372036854774272
n1 -9223372036854774271
n2 -9223372036854774270
n3 -9223372036854774269

p  0x7fffffffffffffff
n  0x8000000000000601
n0 -9223372036854774271
n1 -9223372036854774270
n2 -9223372036854774269
n3 -9223372036854774268

mlr --oxtab --ofmt %.8g put $n0=$n-0;$n1=$n-1;$n2=$n-2;$n3=$n-3 ./reg-test/input/int64arith.dkvp
p  0x7ffffffffffff9ff
n  0x8000000000000000
n0 -9223372036854775808
n1 -9.223372e+18
n2 -9.223372e+18
n3 -9.223372e+18

p  0x7ffffffffffffa00
n  0x8000000000000001
n0 -9223372036854775807
n1 -9223372036854775808
n2 -9.223372e+18
n3 -9.223372e+18

p  0x7ffffffffffffbff
n  0x8000000000000200
n0 -9223372036854775296
n1 -9223372036854775297
n2 -9223372036854775298
n3 -9223372036854775299

p  0x7ffffffffffffc00
n  0x8000000000000400
n0 -9223372036854774784
n1 -9223372036854774785
n2 -9223372036854774786
n3 -9223372036854774787

p  0x7ffffffffffffe00
n  0x8000000000000401
n0 -9223372036854774783
n1 -9223372036854774784
n2 -9223372036854774785
n3 -9223372036854774786

p  0x7ffffffffffffffe
n  0x8000000000000600
n0 -9223372036854774272
n1 -9223372036854774273
n2 -9223372036854774274
n3 -9223372036854774275

p  0x7fffffffffffffff
n  0x8000000000000601
n0 -9223372036854774271
n1 -9223372036854774272
n2 -9223372036854774273
n3 -9223372036854774274

mlr --oxtab --ofmt %.8g put $n0=$n*0;$n1=$n*1;$n2=$n*2;$n3=$n*3 ./reg-test/input/int64arith.dkvp
p  0x7ffffffffffff9ff
n  0x8000000000000000
n0 0
n1 -9.223372e+18
n2 -1.8446744e+19
n3 -2.7670116e+19

p  0x7ffffffffffffa00
n  0x8000000000000001
n0 0
n1 -9.223372e+18
n2 -1.8446744e+19
n3 -2.7670116e+19

p  0x7ffffffffffffbff
n  0x8000000000000200
n0 0
n1 -9.223372e+18
n2 -1.8446744e+19
n3 -2.7670116e+19

p  0x7ffffffffffffc00
n  0x8000000000000400
n0 0
n1 -9223372036854774784
n2 -1.8446744e+19
n3 -2.7670116e+19

p  0x7ffffffffffffe00
n  0x8000000000000401
n0 0
n1 -9223372036854774783
n2 -1.8446744e+19
n3 -2.7670116e+19

p  0x7ffffffffffffffe
n  0x8000000000000600
n0 0
n1 -9223372036854774272
n2 -1.8446744e+19
n3 -2.7670116e+19

p  0x7fffffffffffffff
n  0x8000000000000601
n0 0
n1 -9223372036854774271
n2 -1.8446744e+19
n3 -2.7670116e+19

